src/languages/latex.js

```diff
@@ -1,70 +1,243 @@
+    variants: DOUBLE_CARET_VARIANTS
+  };
+  const SPECIAL_CATCODE = {
+    className: 'built_in',
+    relevance: 0,
+    begin: /[$&^_]/
+  };
+  const MAGIC_COMMENT = {
+    className: 'meta',
+    begin: '% !TeX',
+    end: '$',
+    relevance: 10
+  };
+  const COMMENT = hljs.COMMENT(
+    '%',
+    '$',
+    {
+      relevance: 0
+    }
+  );
+  const EVERYTHING_BUT_VERBATIM = [
+    CONTROL_SEQUENCE,
+    MACRO_PARAM,
+    DOUBLE_CARET_CHAR,
+    SPECIAL_CATCODE,
+    MAGIC_COMMENT,
+    COMMENT
+  ];
+  const BRACE_GROUP_NO_VERBATIM = {
+    begin: /\{/, end: /\}/,
+    relevance: 0,
+    contains: ['self', ...EVERYTHING_BUT_VERBATIM]
+  };
+  const ARGUMENT_BRACES = hljs.inherit(
+    BRACE_GROUP_NO_VERBATIM,
+    {
+      relevance: 0,
+      endsParent: true,
+      contains: [BRACE_GROUP_NO_VERBATIM, ...EVERYTHING_BUT_VERBATIM]
+    }
+  );
+  const ARGUMENT_BRACKETS = {
+    begin: /\[/,
+      end: /\]/,
+    endsParent: true,
+    relevance: 0,
+    contains: [BRACE_GROUP_NO_VERBATIM, ...EVERYTHING_BUT_VERBATIM]
+  };
+  const END_PARENT = {
+    begin: /(?=[.$])/,
+    end: /(?=[.$])/,
+    endsParent: true,
+    relevance: 0
+  };
+  const SPACE_GOBBLER = {
+    begin: /\s+/,
+    relevance: 0
+  };
+  const ARGUMENT_M = [ARGUMENT_BRACES];
+  const ARGUMENT_O = [ARGUMENT_BRACKETS, END_PARENT];
+  const ARGUMENT_AND_THEN = function(arg, starts_mode) {
+    return {
+      contains: [SPACE_GOBBLER],
+      starts: {
+        relevance: 0,
+        contains: arg,
+        starts: starts_mode
+      }
+    };
+  };
+  const CSNAME = function(csname, starts_mode) {
+    return {
+        begin: '\\\\' + csname + '(?![a-zA-Z@:_])',
+        keywords: {$pattern: /\\[a-zA-Z]+/, keyword: '\\' + csname},
+        relevance: 0,
+        contains: [SPACE_GOBBLER],
+        starts: starts_mode
+      };
+  };
+  const BEGIN_ENV = function(envname, starts_mode) {
+    return hljs.inherit(
       {
-        className: 'formula',
-        contains: [COMMAND],
+        begin: '\\\\begin(?=\\s*\\r?\\n?\\s*\\{' + envname + '\\})',
+        keywords: {$pattern: /\\[a-zA-Z]+/, keyword: '\\begin'},
         relevance: 0,
-        variants: [
-          {begin: /\$\$/, end: /\$\$/},
-          {begin: /\$/, end: /\$/}
-        ]
       },
-      hljs.COMMENT(
-        '%',
-        '$',
-        {
-          relevance: 0
-        }
+      ARGUMENT_AND_THEN(ARGUMENT_M, starts_mode)
+    );
+  };
+  const VERBATIM_DELIMITED_EQUAL = (innerName = "string") => {
+    return hljs.END_SAME_AS_BEGIN({
+      className: innerName,
+      begin: /(.|\r?\n)/,
+      end: /(.|\r?\n)/,
+      excludeBegin: true,
+      excludeEnd: true,
+      endsParent: true
+    });
+  };
+  const VERBATIM_DELIMITED_ENV = function(envname) {
+    return {
+      className: 'string',
+      end: '(?=\\\\end\\{' + envname + '\\})'
+    };
+  };
+
+  const VERBATIM_DELIMITED_BRACES = (innerName = "string") => {
+    return {
+      relevance: 0,
+      begin: /{/,
```

Comment:
```suggestion
      begin: /\{/,
```

Needs to be escaped.

Reply:
Yes, and the `end` a couple of lines down, too.